main.go

```diff
@@ -200,7 +200,7 @@ func run(state overseer.State) {
 			c.Filter = filter
 		}
 
-		if err = e.ScanGit(ctx, sources.NewConfig(g)); err != nil {
+		if err = e.ScanGit(ctx, *sources.NewConfig(g)); err != nil {
```

Comment:
**Suggestion:** make `sources.NewConfig` return the struct instead of a pointer so we don't have to deref here.

Reply:
yea i had thought about going that route. I guess I left it this way since it was the standard way that everyone is used to seeing this functional options implemented. One of those things where when you take a quick glance you just kinda go, "oh its doing that" and almost automatically think it should return a pointer. Similar to how we are used to seeing `err != nil` and when we see `err == nil` it usually requires a double take.

That being said I don't have a strong personal pref on this.

Reply:
Cool, thanks for sharing your reasoning! I'm not familiar with that pattern so it didn't occur to me.